src/compiler/crystal/compiler.cr

```diff
@@ -240,7 +240,7 @@ module Crystal
 
     private def codegen_single_unit(unit, target_triple, multithreaded)
       unit.llvm_mod.target = target_triple
-      ifdef x86_64
+      if target_triple =~ /^x86_64/
```

Comment:
Currently if you have a 64-bit static build of the Crystal compiler binary, then it never uses ``DataLayout32`` regardless of what target triple is set via the command line because the relevant code branch is simply not compiled in. Replacing 'ifdef' with 'if' allows to make the selection between the use of ``DataLayout32`` and ``DataLayout64`` at runtime.